release_unpack_macos fails on Xcode 27

1. `flutter create new_app && cd new_app`
2. `open macos/Runner.xcworkspace/`
3. Change `MACOSX_DEPLOYMENT_TARGET` to macOS 12.0 in the build setting for both the `Runner` target and `Flutter Assemble` target (this is to bypass https://github.com/flutter/flutter/issues/187762).
4. `flutter build macos`

```
Target release_unpack_macos failed: Exception: Binary /Users/vashworth/Development/experiment/new_app/build/macos/Build/Products/Release/FlutterMacOS.framework/Versions/A/FlutterMacOS does not contain architectures "arm64 x86_64".

lipo -info:
Architectures in the fat file: /Users/vashworth/Development/experiment/new_app/build/macos/Build/Products/Release/FlutterMacOS.framework/Versions/A/FlutterMacOS are: x86_64 arm64
Failed to copy Flutter framework.
** BUILD FAILED **
```

--- a/packages/flutter_tools/lib/src/build_system/targets/darwin.dart
+++ b/packages/flutter_tools/lib/src/build_system/targets/darwin.dart
@@ -80,20 +80,22 @@ abstract class UnpackDarwin extends Target {
     ]);
     final lipoInfo = infoResult.stdout as String;
 
-    final ProcessResult verifyResult = await environment.processManager.run(<String>[
-      'lipo',
-      frameworkBinaryPath,
-      '-verify_arch',
-      ...archList,
-    ]);
+    for (final arch in archList) {
+      final ProcessResult verifyResult = await environment.processManager.run(<String>[
+        'lipo',
+        frameworkBinaryPath,
+        '-verify_arch',
+        arch,
+      ]);
 
-    if (verifyResult.exitCode != 0) {
-      throw Exception(
-        'Binary $frameworkBinaryPath does not contain architectures "$archs".\n'
-        '\n'
-        'lipo -info:\n'
-        '$lipoInfo',
-      );
+      if (verifyResult.exitCode != 0) {
+        throw Exception(
+          'Binary $frameworkBinaryPath does not contain architecture "$arch" (expected "$archs").\n'
+          '\n'
+          'lipo -info:\n'
+          '$lipoInfo',
+        );
+      }
     }
 
     // Skip thinning for non-fat executables.

--- a/packages/flutter_tools/test/general.shard/build_system/targets/ios_test.dart
+++ b/packages/flutter_tools/test/general.shard/build_system/targets/ios_test.dart
@@ -1030,10 +1030,7 @@ void main() {
           command: <String>['lipo', '-info', binary.path],
           stdout: 'Architectures in the fat file:',
         ),
-        FakeCommand(
-          command: <String>['lipo', binary.path, '-verify_arch', 'arm64', 'x86_64'],
-          exitCode: 1,
-        ),
+        FakeCommand(command: <String>['lipo', binary.path, '-verify_arch', 'arm64'], exitCode: 1),
       ]);
 
       await expectLater(
@@ -1043,7 +1040,7 @@ void main() {
             (Exception exception) => exception.toString(),
             'description',
             contains(
-              'does not contain architectures "arm64 x86_64".\n\n'
+              'does not contain architecture "arm64" (expected "arm64 x86_64").\n\n'
               'lipo -info:\nArchitectures in the fat file:',
             ),
           ),
@@ -1070,7 +1067,8 @@ void main() {
           command: <String>['lipo', '-info', binary.path],
           stdout: 'Architectures in the fat file:',
         ),
-        FakeCommand(command: <String>['lipo', binary.path, '-verify_arch', 'arm64', 'x86_64']),
+        FakeCommand(command: <String>['lipo', binary.path, '-verify_arch', 'arm64']),
+        FakeCommand(command: <String>['lipo', binary.path, '-verify_arch', 'x86_64']),
         FakeCommand(
           command: <String>[
             'lipo',
@@ -1234,7 +1232,8 @@ void main() {
           command: <String>['lipo', '-info', binary.path],
           stdout: 'Architectures in the fat file:',
         ),
-        FakeCommand(command: <String>['lipo', binary.path, '-verify_arch', 'arm64', 'x86_64']),
+        FakeCommand(command: <String>['lipo', binary.path, '-verify_arch', 'arm64']),
+        FakeCommand(command: <String>['lipo', binary.path, '-verify_arch', 'x86_64']),
         FakeCommand(
           command: <String>[
             'lipo',

--- a/packages/flutter_tools/test/general.shard/build_system/targets/macos_test.dart
+++ b/packages/flutter_tools/test/general.shard/build_system/targets/macos_test.dart
@@ -251,10 +251,7 @@ void main() {
       processManager.addCommands(<FakeCommand>[
         copyFrameworkCommand,
         lipoInfoFatCommand,
-        FakeCommand(
-          command: <String>['lipo', binary.path, '-verify_arch', 'arm64', 'x86_64'],
-          exitCode: 1,
-        ),
+        FakeCommand(command: <String>['lipo', binary.path, '-verify_arch', 'arm64'], exitCode: 1),
       ]);
 
       await expectLater(
@@ -264,7 +261,7 @@ void main() {
             (Exception exception) => exception.toString(),
             'description',
             contains(
-              'does not contain architectures "arm64 x86_64".\n\nlipo -info:\nArchitectures in the fat file:',
+              'does not contain architecture "arm64" (expected "arm64 x86_64").\n\nlipo -info:\nArchitectures in the fat file:',
             ),
           ),
         ),

